imaginairy/schema.py

```diff
@@ -1,155 +1,116 @@
-        self._load_img()
-        return getattr(self._img, key)
-
-    def __setstate__(self, state):
-        self.__dict__.update(state)
-
-    def __getstate__(self):
-        return self.__dict__
-
-    def _load_img(self):
-        if self._img is None:
-            from PIL import Image, ImageOps
-
-            if self._lazy_filepath:
-                self._img = Image.open(self._lazy_filepath)
-                logger.debug(
-                    f"Loaded input 🖼  of size {self._img.size} from {self._lazy_filepath}"
+        self,
+        *,
+        filepath: str | Path | None = None,
+        url: str | None = None,
+        img: Image.Image | None = None,
+        b64: str | None = None,
+    ) -> None:
+        match sum(param is not None for param in [filepath, url, img, b64]):
+            case 0:
+                raise ValueError(
+                    "You must specify a url or filepath or img or base64 string"
                 )
-            elif self._lazy_url:
-                import requests
+            case s if s > 1:
+                raise ValueError("You cannot multiple input methods")
+            case _:
+                ...
 
-                self._img = Image.open(
-                    BytesIO(
-                        requests.get(self._lazy_url, stream=True, timeout=60).content
-                    )
-                )
+        if filepath is not None:
+            self.data = self._parse_filepath(filepath)
 
-                logger.debug(
-                    f"Loaded input 🖼  of size {self._img.size} from {self._lazy_url}"
-                )
-            else:
-                raise ValueError("You must specify a url or filepath")
-            # fix orientation
-            self._img = ImageOps.exif_transpose(self._img)
+        if url is not None:
+            self.data = self._parse_url(url)
+
+        if img is not None:
+            self.data = img
+
+        if b64 is not None:
+            self.data = self._parse_base64(b64)
+
+    # def __getattr__(self, key: str) -> Any:
+    #     if not key == "data":
+    #         self._load_data()
+    #     return super().__getattribute__(key)
+
+    # def __setstate__(self, state: dict[str, Any]) -> None:
+    #     self.__dict__.update(state)
+
+    # def __getstate__(self) -> dict[str, Any]:
+    #     return self.__dict__
+
+    def __str__(self) -> str:
+        return self.as_base64()
+
+    def __repr__(self) -> str:
+        match self.data:
+            case Path():
+                return f"LazyLoadingImage(filepath={self.data})"
+            case Url():
+                return f"LazyLoadingImage(url={self.data})"
+            case Image.Image():
+                return f"LazyLoadingImage(img={self.data})"
 
     @classmethod
-    def __get_pydantic_core_schema__(
-        cls, source_type: Any, handler: GetCoreSchemaHandler
-    ) -> core_schema.CoreSchema:
+    def __get_pydantic_core_schema__(cls, *_: Any) -> core_schema.CoreSchema:
```

Comment:
__get_pydantic_core_schema__ was the most confusing to me when I was migrating to  .  Was a battle to get pydantic 2.0 and the "image" type all working with the FastAPI server.  Awesome if you figured out how to simplify that.